### ISSUE TITLE
Intersecting geohashes at zero meridian, equator

### ISSUE THREAD

**Issue by jstaffans** [Factual/geo#1]
Seems there is a problem around around the equator and the zero meridian:

``` clojure
(def factory (GeometryFactory. (PrecisionModel.) 4326))

(def geometry 
  (.toGeometry 
    factory 
    (Envelope. 2.0 6.0 42.0 48.0)))

(time (geohashes-intersecting geometry 10))
"Elapsed time: 1.296939 msecs"
=> (...)  

(def geometry 
  (.toGeometry 
    factory 
    (Envelope. -2.0 2.0 42.0 48.0)))

(time (geohashes-intersecting geometry 10))
; never finishes!
```

Or am I using the geohashes-intersecting function the wrong way?

**Comment by jstaffans** [comment]
Worked around this by checking if the geometry straddles the zero meridian and splitting it into two if it does.

**Comment by worace** [comment]
@jstaffans I realize this issue is ancient and I'm sure you have moved on to other things but for what it's worth I finally released a new version `1.1.0` of `factual/geo` which should fix this issue.

here's a test-case i included based on your report: https://github.com/Factual/geo/blob/master/test/geo/t_geohash.clj#L170-L175

**Comment by jstaffans** [comment]
Thanks @worace!
